Mage/Mage.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+using Newtonsoft.Json.Linq;
+
+using UnityEngine;
+
+public class Mage : Singleton<Mage> {
+	//
+	public EventManager eventManager;
+	public RPCClient rpcClient;
+	public MessageStream messageStream;
+	public Archivist archivist;
+
+	private ConsoleWriter _consoleWriter;
+	private Logger _logger;
+
+
+	//
+	private Dictionary<string, Logger> _loggers = new Dictionary<string, Logger>();
+	public Logger logger (string context = null) {
+		if (string.IsNullOrEmpty(context)) {
+			context = "Default";
+		}
+
+		if (_loggers.ContainsKey(context)) {
+			return _loggers[context];
+		}
+
+		Logger newLogger = new Logger (context);
+		_loggers.Add (context, new Logger(context));
+		return newLogger;
+	}
+
+
+	//
+	public Mage() {
+		eventManager = new EventManager ();
+		rpcClient = new RPCClient ();
+		messageStream = new MessageStream();
+		archivist = new Archivist ();
+
+		_consoleWriter = new ConsoleWriter ();
+		_logger = logger("mage");
+	}
+
+	//
+	public void setEndpoints (string baseUrl, string appName, string username = null, string password = null) {
+		rpcClient.setEndpoint (baseUrl, appName, username, password);
+		messageStream.setEndpoint (baseUrl, username, password);
+	}
+
+	//
+	public void setup (List<string> moduleNames, Action<Exception> cb) {
+		_logger.info ("Setting up modules");
+
+		Async.each<string> (moduleNames, (string moduleName, Action<Exception> callback) => {
+			_logger.info("Setting up module: " + moduleName);
+
+			// Use reflection to find module by name
+			Assembly assembly = Assembly.GetExecutingAssembly();
+			Type[] assemblyTypes = assembly.GetTypes();
```

Comment:
funnily enough this worked for TJ project just fine, even on mobile hmmm will take a look anyway